### PR TITLE
Remove type parameters from WorkQueue.

### PR DESCRIPTION
**Pull request by Ms2ger**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9050)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Extracted from #7524.

r? @pcwalton

File: components/util/workqueue.rs [hunk 1]
```diff
@@ -190,22 +179,31 @@ impl<QueueData: Sync, WorkData: Send> WorkerThread<QueueData, WorkData> {
...
+unsafe fn erase_proxy_type<QueueData>(t: Box<FnBox(&mut WorkerProxy<QueueData>) + Send>)
+                                     -> Box<FnBox(&mut WorkerProxy<c_void>) + Send> {
+    transmute(t)
+}
+
 /// A handle to the work queue that individual work units have.
-pub struct WorkerProxy<'a, QueueData: 'a, WorkData: 'a> {
-    worker: &'a mut Worker<WorkUnit<QueueData, WorkData>>,
+pub struct WorkerProxy<'a, QueueData: 'a> {
+    worker: &'a mut Worker<Box<FnBox(&mut WorkerProxy<c_void>) + Send>>,
```
Comment (pcwalton) [inline components/util/workqueue.rs]: Is this going to result in extra allocations per work unit? This code is designed to not allocate for those.

**Comment by jdm** [comment]
@Ms2ger This is waiting on a response from you.

**Comment by Ms2ger** [comment]
This is all @eefriedman, but it seems plausible.

**Comment by jdm** [comment]
Nobody's championing this work, so no need to leave it open. #9959 is tracking this improvement.

**Pull request closed without merging**